filebeat/input/filestream/input_integration_test.go

```diff
@@ -0,0 +1,144 @@
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package filestream
+
+import (
+	"context"
+	"runtime"
+	"sync"
+	"testing"
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+// test_close_renamed from test_harvester.py
+func TestFilestreamCloseRenamed(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("renaming files while Filebeat is running is not supported on Windows")
+	}
+
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp, err := (&loginp.InputManager{
+		Logger:     logp.L(),
+		StateStore: env.stateStore,
+		Type:       pluginName,
+		Configure:  configure,
+	}).Create(common.MustNewConfigFrom(
+		map[string]interface{}{
+			"paths":                                []string{env.abspath(testlogName) + "*"},
+			"prospector.scanner.check_interval":    "1ms",
+			"close.on_state_change.check_interval": "1ms",
+			"close.on_state_change.renamed":        "true",
+		},
+	))
+
+	if err != nil {
+		t.Fatalf("cannot create filestream input: %+v", err)
+	}
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	inputCtx := input.Context{Logger: logp.L(), Cancelation: ctx}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		inp.Run(inputCtx, env.pipeline)
+	}()
+
+	testlines := []byte("first log line\n")
+	env.mustWriteLinesToFile(testlogName, testlines)
+
+	// first event has made it successfully
+	env.waitUntilEventCount(1)
+	// check registry
+	env.requireOffsetInRegistry(testlogName, len(testlines))
+
+	testlogNameRotated := "test.log.rotated"
+	env.mustRenameFile(testlogName, testlogNameRotated)
+
+	newerTestlines := []byte("new first log line\nnew second log line\n")
+	env.mustWriteLinesToFile(testlogName, newerTestlines)
+
+	// new two events arrived
+	env.waitUntilEventCount(3)
+
+	cancelInput()
+	wg.Wait()
+
+	env.requireOffsetInRegistry(testlogNameRotated, len(testlines))
+	env.requireOffsetInRegistry(testlogName, len(newerTestlines))
+}
+
+// test_close_eof from test_harvester.py
+func TestFilestreamCloseEOF(t *testing.T) {
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp, _ := (&loginp.InputManager{
+		Logger:     logp.L(),
+		StateStore: env.stateStore,
+		Type:       pluginName,
+		Configure:  configure,
+	}).Create(common.MustNewConfigFrom(
+		map[string]interface{}{
+			"paths":                             []string{env.abspath(testlogName)},
+			"prospector.scanner.check_interval": "24h",
+			"close.reader.on_eof":               "true",
+		},
+	))
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	inputCtx := input.Context{Logger: logp.L(), Cancelation: ctx}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		inp.Run(inputCtx, env.pipeline)
+	}()
```

Comment:
Can we abstract this pattern for starting and running a configured input as well?

The `Create` + running boilerplate is mostly here in order to be able to run an input instance in the background. It would be nice to express it like:

```
env.MustStartInput(map[string]interface{}{
})
```

The `env` would take case of stopping active inputs + waiting. If we want to support stopping early we could add a context parameter to `MustStartInput`.